### ISSUE TITLE
Restore weights to restore all layers variable

### ISSUE THREAD

**Issue by aymericdamien** [tflearn/tflearn#153]
(Following gitter talk)
Currently, when loading a DNN model, `load` accept a 'weight_only' that enable loading all trainable variables only and not variables used at training time only (moving mean, training step, etc...). However, some layer are using variables that are not trainable (such as batch norm), so these variables will not be restored.
